app/lib/core/view_models/base_view_model.dart

```diff
@@ -1,11 +1,26 @@
-import 'package:app/core/enums/view_state.dart';
+import 'package:app/core/models/user_info.dart';
+import 'package:app/core/services/authentication/auth_service.dart';
+import 'package:app/core/services/navigation/nav_service.dart';
+import 'package:app/core/services/service_locator.dart';
 import 'package:flutter/material.dart';
 
 class BaseViewModel extends ChangeNotifier {
-  ViewState _state = ViewState.Idle;
-  ViewState get state => _state;
-  void setState(ViewState viewState) {
-    _state = viewState;
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  UserInformation get currentUser => _authenticationService.currentUser;
```

Comment:
Does every change notifier need the current user? This smells.

Reply:
You need to know if there is a current user in most of viewmodels that communicates to firestore services. I put it in the base model, because we will have multiple screen that  needs to know the current user. A great example would be the AddHistoryViewModel

Reply:
So... that seems anti-pattern to me. Some classes don't need this information, so we shouldn't have it "globally" available to all extensions.

We use the dependency injection service so we don't have to have dependencies in the base class like this.

- [ ] Make it so only the classes that need the service have the service

Reply:
That is indeed a really good point! I'll change this in my next commit.